code/modules/atmospherics/gasmixtures/gas_mixture.dm

```diff
@@ -357,32 +351,36 @@ GLOBAL_LIST_INIT(gaslist_cache, init_gaslist_cache())
 	var/abs_moved_moles = 0
 
 	//GAS TRANSFER
-	for(var/id in sharer_gases - cached_gases) // create gases not in our cache
-		ADD_GAS(id, gases)
+	for(var/new_gas_id in sharer_gases - cached_gases) // create gases not in our cache
+		ADD_GAS(new_gas_id, cached_gases)
 	for(var/id in cached_gases) // transfer gases
-		ASSERT_GAS(id, sharer)
+		ASSERT_GAS(id, sharer_gases)
 
 		var/gas = cached_gases[id]
 		var/sharergas = sharer_gases[id]
 
-		var/delta = QUANTIZE(gas[ARCHIVE] - sharergas[ARCHIVE])/(atmos_adjacent_turfs+1) //the amount of gas that gets moved between the mixtures
+		///the amount of gas that gets moved between the mixtures.
+		var/gas_delta = QUANTIZE(gas[ARCHIVE] - sharergas[ARCHIVE]) / (atmos_adjacent_turfs + 1)
 
-		if(delta && abs_temperature_delta > MINIMUM_TEMPERATURE_DELTA_TO_CONSIDER)
-			var/gas_heat_capacity = delta * gas[GAS_META][META_GAS_SPECIFIC_HEAT]
-			if(delta > 0)
+		if(gas_delta && temperature_delta > MINIMUM_TEMPERATURE_DELTA_TO_CONSIDER)
+			var/gas_heat_capacity = gas_delta * gas[GAS_META][META_GAS_SPECIFIC_HEAT]
+			if(gas_delta > 0)
 				heat_capacity_self_to_sharer += gas_heat_capacity
 			else
 				heat_capacity_sharer_to_self -= gas_heat_capacity //subtract here instead of adding the absolute value because we know that delta is negative.
 
-		gas[MOLES] -= delta
-		sharergas[MOLES] += delta
-		moved_moles += delta
-		abs_moved_moles += abs(delta)
+		gas[MOLES] -= gas_delta
+		sharergas[MOLES] += gas_delta
+		moved_moles += gas_delta
+		abs_moved_moles += abs(gas_delta)
 
 	last_share = abs_moved_moles
 
 	//THERMAL ENERGY TRANSFER
-	if(abs_temperature_delta > MINIMUM_TEMPERATURE_DELTA_TO_CONSIDER)
+	if(temperature_delta > MINIMUM_TEMPERATURE_DELTA_TO_CONSIDER)
+		var/old_self_heat_capacity = heat_capacity(ARCHIVE)
+		var/old_sharer_heat_capacity = sharer.heat_capacity(ARCHIVE)
+
```

Comment:
It significantly changes behavior. The heat capacity of a share is not the same as the heat capacity of a series of shares sampled at any time